hefquin-engine/src/main/java/se/liu/ida/hefquin/engine/federation/access/TPFResponse.java

```diff
@@ -78,8 +78,8 @@ default int getMetadataSize(){
 
 	/**
 	 * Returns the cardinality estimate provided as metadata in the given
-	 * TPF response. Returns <code>null</code> if there is no metadata
-	 * with a cardinality estimate.
+	 * TPF response. Throws CardinalityEstimationUnavailableErrorn if there is
+	 * no metadata with a cardinality estimate.
```

Comment:
I am sorry for the confusion. [My comment](https://github.com/LiUSemWeb/HeFQUIN/issues/419#issuecomment-2812505718) related to this point was not sufficiently detailed. It wasn't my intention to change the behavior of this method. In contrast, the method should indeed return `null` if the TPF response does not contain metadata with a cardinality estimate. The place where the new type of exception (`CardinalityEstimationUnavailableError`) should be created is in `FunctionToObtainCardinalityResponseFromTPFResponse` within `FederationAccessManagerBase1`, at the point when the `getCardinalityEstimate()` function is called and returns `null`.